python/whylogs/api/store/profile_store.py

```diff
@@ -12,11 +12,11 @@ def list(self) -> List[str]:
         pass
 
     @abstractmethod
-    def get(self, query: DateQuery) -> DatasetProfileView:
+    def get(self, query: Union[DateQuery, DatasetIdQuery]) -> DatasetProfileView:
```

Comment:
Perhaps a base class rather than a union?

Reply:
in here i've tried to avoid [dataclass inheritance](https://stackoverflow.com/questions/51575931/class-inheritance-in-python-3-7-dataclasses). but if you think it's still better to inherit, I can make them regular classes :)